tests/integ/modin/index/test_equals.py

```diff
@@ -10,36 +10,81 @@
 from tests.integ.modin.sql_counter import SqlCounter, sql_count_checker
 
 
+def to_snow_index(native_index: native_pd.Index, convert_to_lazy: bool):
+    """
```

Comment:
do we have any test that test what happens if the data for datetime index is query compiler and the type is not timestamp?

Reply:
Good point. Let me add a test with assert check.